src/ray/raylet/raylet_client.h

```diff
@@ -156,16 +125,24 @@ class RayletClient {
   const ResourceMappingType &GetResourceIDs() const { return resource_ids_; }
 
  private:
-  const UniqueID client_id_;
+  /// A mutex to protect the stateful connection.
+  std::mutex mutex_;
+  /// The ID for this raylet client.
+  const ClientID client_id_;
+  /// Whether this client belongs to a worker.
   const bool is_worker_;
+  /// The TaskID for the driver.
   const JobID driver_id_;
+  /// The programming language using this C++ client.
   const Language language_;
   /// A map from resource name to the resource IDs that are currently reserved
   /// for this worker. Each pair consists of the resource ID and the fraction
   /// of that resource allocated for this worker.
   ResourceMappingType resource_ids_;
+  /// The Boost IO service.
+  boost::asio::io_service main_service_;
```

Comment:
The IO service here is only for the requirement of asio socket, it's kind of dummy, and is not supposed to be associated with an actual thread right?  in that case could you make it clear in the comments?